### PR TITLE
Sharing api integration tests adding cases

### PR DESCRIPTION
**Pull request by SergioBertolinSG**
Added tests for creation and update of public sharings of folders.

### CODE REVIEW

**Comment by SergioBertolinSG** [comment]
Last test is failing because of this bug: https://github.com/owncloud/core/issues/20464

File: build/integration/features/sharing-v1.feature [hunk 1]
```diff
@@ -46,6 +46,25 @@ Feature: sharing
...
+  Scenario: Creating a new public share of a folder
+   Given user "user0" exists
+    And As an "user0"
+    When creating a public share with
+      | path | FOLDER |
+      | shareType | 3 |
+      | password | publicpw |
+      | expireDate | +3 days |
+      | publicUpload | true |
+      | permissions | 7 |
```
Comment (rullzer) [inline build/integration/features/sharing-v1.feature]: Just FYI: You don't have to set the permissions and public upload.

File: build/integration/features/sharing-v1.feature [hunk 2]
```diff
@@ -46,6 +46,25 @@ Feature: sharing
...
+    Then the OCS status code should be "100"
+    And the HTTP status code should be "200"
+    And Share fields of last share match with
+      | id | A_NUMBER |
+      | permissions | 7 |
+      | expiration | +3 days |
+      | url | AN_URL |
+      | token | A_TOKEN |
```
Comment (rullzer) [inline build/integration/features/sharing-v1.feature]: Would it be hard to check the actual URL? It should contain the same token.

**Comment by rullzer** [comment]
Thanks a lot again @SergioBertolinSG. Beside my question/remark this looks good. :+1:

**Comment by PVince81** [comment]
:+1:

File: build/integration/features/sharing-v1.feature (hunk 1 above)
Comment (SergioBertolinSG) [inline build/integration/features/sharing-v1.feature]: It shouldn't affect the response.

**Pull request merged**